src/docs/build-deploy-and-maintain-apps/deploy-an-application.md

```diff
@@ -0,0 +1,341 @@
+
+For guidance on resource tuning, refer to this [documentation](https://docs.developer.gov.bc.ca/application-resource-tuning/). You can also watch this [video](https://www.youtube.com/watch?v=rkxVZgn9icU&t=14s) for more explanation. Understanding and adjusting these resources according to your application's needs is a critical aspect of deployment preparation.
+
+---
+
+## Deploying the application
+
+### 1. Creating a deployment using OpenShift CLI or Web Console
+
+To deploy an application, you can use the OpenShift CLI (`oc create-app`) or the web console. Both offer the same capabilities and your choice will be dependent on personal preference. More information can be found in the [official OpenShift documentation](https://docs.openshift.com/container-platform/latest/applications/application_life_cycle_management/creating-applications-using-cli.html).
+
+### 2. Configuring resource limits and scaling options
+
+Resource limits and scaling options are an important part of managing application performance and availability. 
+
+OpenShift allows you to set resource limits (CPU and memory) for each Pod in your deployments, which can prevent a Pod from using more resources than necessary. 
+
+Scaling options allow your application to adapt to different load levels by adjusting the number of running Pods.  The best practice is to provide the suitable resources to your application is to monitoring its normal and peak performance, and decide resources quota and number of replicas dependently. 
+
+This is covered in detail in the [official OpenShift documentation](https://docs.openshift.com/container-platform/latest/nodes/clusters/nodes-cluster-resource-configure.html).
+
+### 3. Get the manifest file from live instance to store as infrastructure as code
+
+Infrastructure as Code (IaC) is a best practice for managing and provisioning technology stacks through machine-readable definition files, rather than physical hardware configuration. You can extract the configuration of a live object in OpenShift as a YAML or JSON manifest file and store it in your version control system, for example Git. This practice makes it easy to track changes, replicate configurations, and recover from mishaps. 
+
+---
+
+## Configuring deployment
+
+### 1. Understanding deployment configurations and their components
+
+Deployment Configurations  in OpenShift are a recipe for deploying your application. They consist of the following are considered as most important field:
+
+- **Pod Template:** This defines the desired state of the pods being deployed, including the base image, ports to expose, and environment variables
+- **Replicas:** The number of instances of your application that should be maintained
+- **Selectors:** Labels that identify the pods managed by the deployment configuration
+- **Triggers:** Events that will cause a new deployment, such as a new image build
```

Comment:
better to be clear that only dc and bc have triggers, deployment and sts do not

Reply:
@ShellyXueHan Deployments and STS can have image triggers, its just a bit more complicated https://docs.openshift.com/container-platform/4.12/openshift_images/triggering-updates-on-imagestream-changes.html